dask/dataframe/tests/test_groupby.py

```diff
@@ -2094,6 +2094,35 @@ def test_series_groupby_idxmax_skipna(skipna):
     assert_eq(result_pd, result_dd)
 
 
+def test_groupby_unique():
+    rng = np.random.RandomState(42)
+    df = pd.DataFrame(
+        {"foo": rng.randint(3, size=100), "bar": rng.randint(10, size=100)}
+    )
+    ddf = dd.from_pandas(df, npartitions=10)
+
+    pd_gb = df.groupby("foo")["bar"].unique()
+    dd_gb = ddf.groupby("foo")["bar"].unique()
+
+    assert pd_gb.index.name == dd_gb.index.name == "foo"
+    assert pd_gb.name == dd_gb.name == "bar"
+    # Use explode because each element is a list; equality fails
+    assert_eq(dd_gb.explode(), pd_gb.explode(), check_names=False)
+
+
+def test_groupby_value_counts():
+    rng = np.random.RandomState(42)
+    df = pd.DataFrame(
+        {"foo": rng.randint(3, size=100), "bar": rng.randint(4, size=100)}
+    )
+    ddf = dd.from_pandas(df, npartitions=2)
+
+    pd_gb = df.groupby("foo")["bar"].value_counts()
+    dd_gb = ddf.groupby("foo")["bar"].value_counts()
+    assert_eq(dd_gb, pd_gb, check_names=False)  # passes
+    assert_eq(dd_gb, pd_gb, check_names=True)
```

Comment:
This test is almost passing, but still failing. It passes if `check_names=True`, but fails when `check_names=False`.

Can I have some tips to help avoid this? I think the issue is around re-naming Pandas `MultiIndex`s.

Details: In my test `test_groupby_value_counts` (code), the Dask and Pandas DataFrame have different names for the outermost columns.

``` python
# -> assert_eq(dd_gb, pd_gb, check_names=True)
(Pdb) p pd_gb.index.names
FrozenList(['foo', 'bar'])
(Pdb) p dd_gb.index.compute().names
FrozenList([None, 'bar'])
(Pdb) p dd_gb.compute()
   bar
0  1       9
   2       9
   # [trimmed]
2  2      12
   0       8
   3       8
   1       3
Name: bar, dtype: int64
(Pdb) p pd_gb
foo  bar
0    1       9
     2       9
     # [trimmed]
2    2      12
     0       8
     3       8
     1       3
Name: bar, dtype: int64
(Pdb)
```

Reply:
Fixed in https://github.com/dask/dask/pull/5358/commits/c095dc265287865232558035390849c46f62dbbf hopefully. Needed to pass through `names` to concat.

Reply:
Thank you!